insecure/corruptible/network.go

```diff
@@ -0,0 +1,370 @@
+	flowNetwork           flownet.Network // original flow network of the node.
+	server                *grpc.Server    // touch point of attack network to this factory.
+	gRPCListenAddress     net.Addr
+	conduitFactory        insecure.CorruptibleConduitFactory
+	attackerInboundStream insecure.CorruptibleConduitFactory_ConnectAttackerServer // inbound stream to attacker
+
+	receiptHasher  hash.Hasher
+	spockHasher    hash.Hasher
+	approvalHasher hash.Hasher
+}
+
+var _ flownet.Network = &Network{}
+
+func NewCorruptibleNetwork(
+	logger zerolog.Logger,
+	chainId flow.ChainID,
+	address string,
+	me module.Local,
+	codec flownet.Codec,
+	flowNetwork flownet.Network,
+	conduitFactory insecure.CorruptibleConduitFactory) (*Network, error) {
+	if chainId != flow.BftTestnet {
+		panic("illegal chain id for using corruptible network")
+	}
+
+	corruptibleNetwork := &Network{
+		codec:          codec,
+		me:             me,
+		conduitFactory: conduitFactory,
+		flowNetwork:    flowNetwork,
+		logger:         logger.With().Str("component", "corruptible-network").Logger(),
+		receiptHasher:  utils.NewExecutionReceiptHasher(),
+		spockHasher:    utils.NewSPOCKHasher(),
+		approvalHasher: verutils.NewResultApprovalHasher(),
+	}
+
+	err := corruptibleNetwork.conduitFactory.RegisterEgressController(corruptibleNetwork)
+	if err != nil {
+		return nil, fmt.Errorf("could not register egress controller on conduit factory: %w", err)
+	}
+	corruptibleNetwork.ComponentManager = component.NewComponentManagerBuilder().
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			corruptibleNetwork.flowNetwork.Start(ctx)
+			<-corruptibleNetwork.flowNetwork.Ready()
+
+			ready()
+
+			<-corruptibleNetwork.flowNetwork.Done()
+		}).
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			corruptibleNetwork.start(ctx, address)
+			ready()
+
+			<-ctx.Done()
+			corruptibleNetwork.stop()
+
+		}).Build()
+
+	return corruptibleNetwork, nil
+}
+
+// Register serves as the typical network registration of the given message processor on the channel.
+// Except, it first wraps the given processor around a corruptible message processor, and then
+// registers the corruptible message processor to the original flow network.
+func (n *Network) Register(channel channels.Channel, messageProcessor flownet.MessageProcessor) (flownet.Conduit, error) {
+	corruptibleProcessor := NewCorruptibleMessageProcessor(n.logger, messageProcessor)
+	// TODO: we can dissolve CCF and instead have a decorator pattern to turn a conduit into
+	// a corrupted one?
+	conduit, err := n.flowNetwork.Register(channel, corruptibleProcessor)
+	if err != nil {
+		return nil, fmt.Errorf("could not register corruptible message processor on channel: %s, %w", channel, err)
+	}
+	return conduit, nil
+}
+
+func (n *Network) RegisterBlobService(channel channels.Channel, store datastore.Batching, opts ...flownet.BlobServiceOption) (flownet.BlobService,
```

Comment:
can we add comments for all normal network methods we have to implement that we are NOT corrupting anything but just passing everything through to the flow network? So it's clear what we're changing and what we're leaving as-is

Reply:
https://github.com/onflow/flow-go/pull/2807/commits/cb1644fda89e97c3e29d4d77033a34aaa9757e79